File: problem2.java
// Time Complexity : O(N)
// Space Complexity : O(H)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : no

// isMirror function recursively checks if two subtrees are mirrors of each other.
import javax.swing.tree.TreeNode;

public class problem2 {
    public boolean isSymmetric(TreeNode root) {
        return isMirror(root, root);
    }

    public boolean isMirror(TreeNode t1, TreeNode t2) {
        if (t1 == null && t2 == null) return true;
        if (t1 == null || t2 == null) return false;
        return (t1.val == t2.val)
                && isMirror(t1.right, t2.left)
                && isMirror(t1.left, t2.right);
    }
}

File: Problem2.cs
public class MirrorImageOfItselfTree
    {
        // Time Complexity : O(n)
        // Space Complexity : O(h)
        // Did this code successfully run on Leetcode : Yes
        // Any problem you faced while coding this : No

        bool flag;
        public bool IsSymmetric(TreeNode root)
        {
            flag = true;
            dfs(root.right, root.left);
            return flag;
        }
        public void dfs(TreeNode left, TreeNode right)
        {
            //base case
            if (left == null && right == null)
                return;

            if (left == null || right == null || left.val != right.val)
            {
                flag = false;
                return;
            }

            //Logic
            if (flag)
                dfs(left.left, right.right);
            if (flag)
                dfs(left.right, right.left);
        }
    }
